src/main/java/org/minions/devfund/roy/WhackAMole.java

```diff
@@ -0,0 +1,169 @@
+package org.minions.devfund.roy;
+
+import java.util.Random;
+
+/**
+ * WhackAMole game.
+ */
+public class WhackAMole {
+
+    private int score;
+    private int molesLeft;
+    private int attemptsLeft;
+    private char[][] moleGrid;
+    private static final char MOLE = 'M';
+    private static final char WHACKED_MOLE = 'W';
+    private static final char EMPTY_PLACE = '*';
+
+    /**
+     * Constructor.
+     *
+     * @param numAttempts   to start the game.
+     * @param gridDimension to create the moleGrid for the game.
+     */
+    public WhackAMole(int numAttempts, int gridDimension) {
+        this.moleGrid = new char[gridDimension][gridDimension];
+        attemptsLeft = numAttempts;
+        this.molesLeft = 0;
+        initializeGrid();
+    }
+
+    /**
+     * Verifies if a place is empty or not.
+     *
+     * @param x axis position.
+     * @param y axis position.
+     * @return true if the place is empty otherwise false.
+     */
+    public boolean place(int x, int y) {
+        if (x >= 0 && y >= 0 && x < moleGrid.length && y < moleGrid.length
+                && moleGrid[x][y] == EMPTY_PLACE) {
+            moleGrid[x][y] = MOLE;
+            molesLeft++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Whacks a received position.
+     *
+     * @param x axis position.
+     * @param y axis position.
+     */
+    public void whack(int x, int y) {
+        if (rules() && !giveUp(x, y)) {
+            if (moleGrid[x][y] == MOLE) {
+                moleGrid[x][y] = WHACKED_MOLE;
+                score++;
+                attemptsLeft--;
+                molesLeft--;
+            } else {
+                attemptsLeft--;
+            }
+        } else {
+            printGrid();
+            System.out.println(String.format("Score: %d", score));
+        }
+
+        if (!rules()) {
+            printGrid();
+        }
+    }
+
+    /**
+     * Verifies if the player gives up.
+     *
+     * @param x axis.
+     * @param y axis.
+     * @return true if the player insert (-1, -1) otherwise false.
+     */
+    public boolean giveUp(int x, int y) {
+        return x == -1 && y == -1;
+    }
+
+    /**
+     * Starts the moleGrid with the moles.
+     */
+    public void initializeGrid() {
+        for (int i = 0; i < moleGrid.length; i++) {
+            for (int j = 0; j < moleGrid.length; j++) {
+                moleGrid[i][j] = EMPTY_PLACE;
+            }
+        }
+    }
+
+    /**
+     * Generates random positions to place Moles.
+     */
+    public void placeMoles() {
+        while (molesLeft < moleGrid.length) {
+            int xRandomValue = new Random().nextInt(moleGrid.length);
+            int yRandomValue = new Random().nextInt(moleGrid.length);
+            place(xRandomValue, yRandomValue);
+        }
+    }
+
+    /**
+     * Prints in console the moleGrid without showing the moles but the 'whacked moles'.
+     */
+    public void printGridForUser() {
```

Comment:
Since you created the method gridBuilderForUser
now you can remove this method.